serial_protocol.md

```diff
@@ -186,3 +192,135 @@ Used to stop listening to events at a given path. This must be the same path pro
 	>>BEGIN_STREAM /user/aturing
 	>>END_STREAM /user/aturing
 	<<+OK
+
+#Messaging Protocol
+
+## BEGIN_MESSAGING 
+Called to start communicating with Firebase Cloud Messaging, it requires a server key.
+###Usage
+	BEGIN %Server_Key%
+###Response
+	+OK - Accepted initialization parameters
+###Examples
+	>> BEGIN https://samplechat.firebaseio-demo.com
+	<< +OK
+
+## MESSAGE_TO_USERS
+Called to start composing a message to users by registration id. You can specify as many as you wish separated by spaces on the same line. 
```

Comment:
I guess if we were going one line we would need to do the argument style. So most of it would work out ok:
``` MSG {registration_id=%ID% | registration_ids=%ID%,%ID% | topic=%TOPIC%} high_priority=true|false collapse_key=%collapsekey% dealy_while_idle=true|false time_to_live=0-2419200"

But when we get to the notification message and the data, the body and values of these would need to allow multiline so that would get complicated.